ballerina/types.bal

```diff
@@ -123,5 +144,50 @@ public enum Protocol {
     TLS
 }
 
+# The stream iterator object that is used to iterate through the stream messages.
+isolated class StreamIterator {
+    private boolean isClosed = false;
+
+    # Returns the next message in the stream.
+    #
+    # + return - `record{|Message value;|}` or else `error?` if the stream is closed or any error occurred while retrieving the next message
+    public isolated function next() returns record {|Message value;|}|error? {
+        lock {
+            if self.isClosed {
+                return error Error("Stream is closed. Therefore, no operations are allowed further on the stream.");
+            }
+        }
+        Message|error? result = self.nextResult(self);
```

Comment:
```suggestion
        Message|error? result = self.nextResult();
```

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerina-mqtt/pull/9/commits/56d4c80f1818864c9e0d79f5702f9ccc115eaf87